agent/serviceinfobroker/service_info_broker.go

```diff
@@ -222,9 +222,18 @@ func (sib *ServiceInfoBroker) getPostgreSQLInfo(ctx context.Context, dsn string,
 	var version string
 	if err = db.QueryRowContext(ctx, "SHOW /* agent='serviceinfobroker' */ SERVER_VERSION").Scan(&version); err != nil {
 		res.Error = err.Error()
+		return &res
 	}
-
 	res.Version = version
+
+	var databaseCount int32
+	databaseCountQuery := "SELECT /* agent='serviceinfobroker' */ count(datname) FROM pg_database WHERE datallowconn = true AND datistemplate = false AND datname != current_database()" //nolint:lll
```

Comment:
Just curious why do we not want to count the current database (`datname != current_database()`)?

Reply:
@ademidoff It is aligned with current auto discover query:
https://github.com/percona/pmm/blob/main/managed/services/checks/checks.go#L1184-L1186

Reply:
I just took another look.
https://github.com/percona/postgres_exporter/blob/main/cmd/postgres_exporter/queries.go#L280
https://github.com/percona/pmm/blob/main/managed/services/checks/checks.go#L1184-L1186

Based on two queries from links above I think it should be "merged" query:
SELECT /* agent='serviceinfobroker' */ count(datname) FROM pg_database WHERE datallowconn = true AND datistemplate = false AND datname != current_database() AND has_database_privilege(current_user, datname, 'connect')

@BupycHuk What do you think?

Reply:
1. when we collect number of DBs `datname != current_database()` should be removed, we have it in postgres_exporter to get other DBs existing on the same server, because we already connected to one.
2. regarding `has_database_privilege(current_user, datname, 'connect')`: seems like a good idea, but we need copy it to postgres_exporter as well to not have inconsistency.

Reply:
Ok just give me a several minutes. I will modify this PR and prepare one for postgres_exporter.

Reply:
Thanks @ademidoff for point it out.

Reply:
PR for PG exporter: https://github.com/percona/postgres_exporter/pull/162